authentication/authentication_option.go

```diff
@@ -83,6 +83,22 @@ func WithRetries(maxRetries int, statuses []int) Option {
 	}
 }
 
+// RetryStrategy defines the retry rules that should be followed by the SDK when making requests.
+type RetryStrategy struct {
+	MaxRetries int
+	Statuses   []int
+
+	// PerAttemptTimeout can optionally be set to timeout individual API requests.
+	PerAttemptTimeout time.Duration
+}
+
+// WithRetryStrategy configures the management client to only retry under the conditions provided.
+func WithRetryStrategy(retryStrategy RetryStrategy) Option {
+	return func(a *Authentication) {
+		a.retryStrategy = client.RetryOptions(retryStrategy)
+	}
+}
```

Comment:
Hi @pete-woods ,

Could we update the field mapping in both management and authentication?
```suggestion
func WithRetryStrategy(retryStrategy RetryStrategy) Option {
	return func(a *Authentication) {
		a.retryStrategy = client.RetryOptions{
			MaxRetries:        retryStrategy.MaxRetries,
			Statuses:          retryStrategy.Statuses,
			PerAttemptTimeout: retryStrategy.PerAttemptTimeout,
		}
	}
}
```

**Reasoning:**  
Explicit field mapping is safer and clearer. It properly handles the fact that `authentication.RetryStrategy` and `internal/client.RetryOptions` are distinct types, and avoids relying on direct type casting (`client.RetryOptions(retryStrategy)`), which can be unsafe or misleading when structs evolve independently. 

Thanks!

Reply:
Thanks for the thoughtful reply—that’s a valid perspective, and I agree that compile-time breakage can be a useful signal during development.

That said, since this SDK is public-facing and intended for broader community use, I lean toward **explicit field mapping** for a few reasons:

- It makes the coupling between public and internal types more intentional.
- It improves clarity for contributors unfamiliar with internal implementation details.
- It prevents silent behavioral changes if internal structs evolve with semantics not meant for external exposure.
- It opens the door to validation, defaulting, or transformation logic between layers (if needed later).

Even though it means we might miss a compile-time nudge when fields are added, I think the tradeoff is worth it in an open-source SDK where clarity and safety are priorities.

Let’s move forward with the explicit mapping—thanks again for the great discussion and your contributions!

Reply:
Done!